### DISCUSSION TITLE
How to access content item inside a class?

### DISCUSSION THREAD

**Discussion by alXDraghiciu** [OrchardCMS/OrchardCore#11252] [Q&A]
Hi. What would be the correct way to access data inside a content item to use it in a class or controller? How can I call GraphQL inside the C# code and execute a query?

**Comment by Skrypt** [comment]
You need to use a HttpClient request which will return you a JSON response from the GraphQL API. Then you use System.Text.Json or Newtonsoft.Json to convert JSON back to a C# object.

**Reply by alXDraghiciu** [reply]
I'd like to do it in code not by making a request. Using the ContentManager with the id seems to do the trick.

Is this a correct way?

**Reply by Skrypt** [reply]
Using the ContentManager will do a direct request to the database without passing on the GraphQL endpoint. That's one way among many others to get the data you want. The ContentManager is going to always return you `IEnumerable<ContentItem>` directly if I remember correctly. Though, the REST APIs will return JSON that you need to either parse directly or convert back to C# objects.

**Reply by Skrypt** [reply]
Another way is to use the YesSQL _session directly. That's what the ContentManager uses. And we have Orchard Helpers to call Queries from the OrchardCore.Queries module as @ns8482e suggested.

**Comment by ns8482e** [comment]
If you are running in coupled/decupled on same web application as your cms then you can use `IOrchardHelper` to query  See https://docs.orchardcore.net/en/latest/docs/reference/modules/Queries/#razor-helpers

You can query using GraphQL within same web app or from separate webapp or from spa client. However GraphQL being API endpoint,  you need to use `HttpClient` from C#,  as suggested by @Skrypt
